x-pack/solutions/security/plugins/security_solution/docs/adrs/threat_hunting/0002_timeline_ui_batching_strategy.md

```diff
@@ -0,0 +1,172 @@
+
+This document is only applicable for Timeline datagrid.
+
+## What is a batch
+Batch ( also known as Sample ) in Unified Data table is the collection of max number of documents that are fetched in a single request. When a user queries a timeline for the first time, Batch 0 is fetched and then the user may have the option to fetch further batches on-demand.
+
+
+### Batch v/s Data Grid Pagination
+The concept of batch is different from the Datagrid Pagination. Datagrid Pagination is completely client driven and datagrid may decide how it wants to paginate a particular batch.
+
+For example, let’s say we fetch a batch of 500 documents, now Datagrid may decide to display these 500 documents following configurations 
+10 pages with each page consisting of 50 documents  
+20 pages with each page consisting of 25 documents
+5 pages with each page consisting of 100 documents.
+
+Users will be able to fetch a new batch of documents when they reach the last page.
+
+### Batch Parameters
+Below are the parameters that affect a batch. Any changes to these parameters should trigger a re-fetch of all existing batches or the initial batch depending on the situation.
+
+- **dataViewId**: Currently selected dataView Id in Timeline.
+- **Timerange**: 
+  - **startDate**
+  - **endDate**
+  - **timeRangeKind**
+- **eqlOptions**
+- **indexNames**: List of indices in Timeline.
+- **id**: Timeline Id
+- **fields**:  List of requested fields
+- **filterQuery**:  combined query consisting: 
+  * **KQL/Lucene Query**
+  * **Filters**
+  * **Data Providers**
+- **runtimeMappings**: List of runtime mappings that need to be fetched
+- **language**: KQL | Lucene
+- **limit**: Sample Size or Batch Size
+- **sort**
+
+## Batch fetch/re-fetch Strategy
+Batch fetch/re-fetch strategy depends on the scenario or the action the user is taking. This section will lay out all the scenarios in which a batch should be re-fetched and its corresponding strategy.
+
+### New Batch Requested
+When a user requests a new batch after clicking on `Load more` as shown below, a new batch will be fetched. For example, if user is on Batch 0 and clicks on Load more, a new batch will be fetched with below strategy :
+
+![](./images/load_more_timeline.png)
+
+```js
+{
+  /* Fetch Batch 1 */
+  activePage: 1, /* current batch */
+  querySize: 500 /* batch/sample size */
+}
+```
+
+Consequently, if a user requests a new batch again ( for the 3rd time ), Batch 2 will be fetched with below strategy.
+
+```js
+{
+	/* Fetch Batch 2 */
+	activePage: 2,
+	querySize: 500 /* batch/sample size */
+}
+```
+ 
+### Refresh Button
+
+When a user clicks on the Refresh button in the query tab, it should explicitly re-fetch new data ir-respective of the fact if any batch parameters have been changed or not.
```

Comment:
`ir-respective` => `irrespective`

Reply:
I would also be more specific here, what do we mean by `new data`?

Reply:
made it more clear in fd2ababe4e4